ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -84,26 +86,38 @@
   private final RewrittenAnalysis analysis;
   private final FunctionRegistry functionRegistry;
   private final AggregateAnalyzer aggregateAnalyzer;
+  private final BiFunction<PlanNode, Expression, FilterNode> filterNodeFactory;
   private final ColumnReferenceRewriter refRewriter;
 
   public LogicalPlanner(
       final KsqlConfig ksqlConfig,
       final ImmutableAnalysis analysis,
-      final FunctionRegistry functionRegistry
+      final FunctionRegistry functionRegistry,
+      final BiFunction<PlanNode, Expression, FilterNode> filterNodeFactory
```

Comment:
I think you misunderstood what I was meaning when I was talking about factory classes.  No worries. But I don't think injecting this `filterNodeFactory` is actually used by any testing, and its not what I had in mind, so we can probably simplify the code by removing it and just calling the function directly.